src/controllers/with-router/index.tsx

```diff
@@ -23,20 +27,20 @@ const getWrappedComponentDisplayName = (
   return `withRouter(${componentDisplayName})`;
 };
 
-export const withRouter = <P extends Record<string, any>>(
+export const withRouter = <P extends Record<string, any> = {}>(
   WrappedComponent: ComponentType<P>
-) => {
+): ComponentType<Omit<P, keyof WithRouterProps>> => {
   const displayName = getWrappedComponentDisplayName(WrappedComponent);
-  const Component = WrappedComponent as ComponentType<WithRouter & P>;
-  const ComponentWithRouter = (props: P) => (
+  const Component = WrappedComponent;
+  const ComponentWithRouter = (unknownProps: any) => (
```

Comment:
is "unknown" necessary?

Reply:
no it's not

Reply:
I would just leave it as `props` if possible